src/actions/copy-async.ts

```diff
@@ -38,15 +28,46 @@ async function getOneFile(filepath: string) {
   return undefined;
 }
 
-export type CopyAsyncOptions = CopySingleAsyncOptions & CopyOptions;
+type CopySingleAsyncOptions = Parameters<MemFsEditor['append']>[2] & {
+  append?: boolean;
+
+  /**
+   * Transform both the file path and content during copy.
+   * @param destinationPath The destination file path
+   * @param sourcePath The source file path
+   * @param contents The file content as Buffer
+   * @returns Tuple of [new filepath, new content]
+   */
+  fileTransform?: (
+    destinationPath: string,
+    sourcePath: string,
+    contents: Buffer,
+  ) => [string, string | Buffer] | Promise<[string, string | Buffer]>;
```

Comment:
I wonder if we should return an object instead of a tuple.

Reply:
We could, no strong feeling either interface personally. I went with a tuple because it was a "path + contents" pair.